electrumx/server/block_processor.py

```diff
@@ -556,7 +519,89 @@ def validate_ft_rules_raw_tx(self, raw_tx):
                 f'ft_output_blueprint.outputs={outputs}\n'
                 f'ft_output_blueprint.fts_burned={fts_burned}'
             )
-    
+        return AtomicalsValidation(
+            raw_tx,
+            tx_hash,
+            operations_found_at_inputs,
+            encode_atomical_ids_hex(atomicals_spent_at_inputs),
+            auto_encode_bytes_items(encode_atomical_ids_hex(ft_output_blueprint)),
+        )
+
+    # Helper method to decode the PSBT and returns formatted structure.
+    def transaction_decode_psbt_blueprint(self, psbt_hex: str) -> dict:
+        raw_tx = psbt.PartiallySignedTransaction.from_base64_or_binary(
```

Comment:
Personally, I feel that manually parsing the PSBT structure is better, and there's no need to introduce additional libraries.